src/java.base/share/classes/java/lang/foreign/GroupLayout.java

```diff
@@ -73,4 +76,136 @@ public sealed interface GroupLayout extends MemoryLayout permits StructLayout, U
      */
     @Override
     GroupLayout withByteAlignment(long byteAlignment);
+
+    /**
+     * {@return a {@link Function} that can project {@linkplain MemorySegment MemorySegments} onto new instances
+     * of the provided {@link Record} {@code type} by means of matching the names of the
+     * record components with the names of the elements in this group layout}
+     * <p>
+     * The mapping between the record type and this memory layout is defined as follows;
+     * <p>
+     * Let <em>R</em> be the provided record {@code type} with the constituent components
+     * <em>C<sub>0</sub></em>, ..., <em>C<sub>N-1</sub></em>, where <em>N</em> is non-negative.
```

Comment:
general stylistic note - you some fancy html here to get close to some math notation. This is nice, but is also inconsistent with what has been done in other parts of the memory layout API - see e.g. MemoryLayout::var handle, which just uses `A_0`, `A_1` etc.